File: frontend/Steiner/Models/Asistencia.swift
import Foundation

public struct AsistenciaCreate: Codable {
    public let tema: String
    public let fecha_clase: String // ISO8601 string
    public let id_clase: Int
    public let id_alumno: Int
}


public struct AsistenciaOut: Codable, Identifiable {
    public var id: Int { id_alumno * 10000 + id_clase } // ID generado
    public let tema: String
    public let fecha_clase: String
    public let id_clase: Int
    public let id_alumno: Int
    
    enum CodingKeys: String, CodingKey {
        case tema, fecha_clase, id_clase, id_alumno
    }
}

File: backend/app/schemas/asistencia.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class AsistenciaBase(BaseModel):
    tema: str
    fecha_clase: datetime

class AsistenciaCreate(AsistenciaBase):
    id_clase: int
    id_alumno: int

class AsistenciaUpdate(AsistenciaBase):
    tema: Optional[str]
    fecha_clase: Optional[datetime]

class AsistenciaOut(AsistenciaBase):
    id_clase: int
    id_alumno: int

    model_config = {"from_attributes": True}

class AsistenciaOutWithUsers(AsistenciaBase):
    nombre: str
    clase: str
    
    model_config = {"from_attributes": True}
